src/test/javascript/spec/component/tutorial-groups/crud/tutorial-group-form.component.spec.ts

```diff
@@ -0,0 +1,122 @@
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { ArtemisTranslatePipe } from 'app/shared/pipes/artemis-translate.pipe';
+import { MockPipe, MockProvider } from 'ng-mocks';
+import { TutorialGroupFormComponent, TutorialGroupFormData } from 'app/course/tutorial-groups/crud/tutorial-group-form/tutorial-group-form.component';
+import { CourseManagementService } from 'app/course/manage/course-management.service';
+import { AlertService } from 'app/core/util/alert.service';
+import { of } from 'rxjs';
+import { CourseGroup } from 'app/entities/course.model';
+import { HttpResponse } from '@angular/common/http';
+import { User } from 'app/core/user/user.model';
+import { NgbTypeaheadModule } from '@ng-bootstrap/ng-bootstrap';
+
+describe('TutorialGroupFormComponent', () => {
+    let tutorialGroupFormComponentFixture: ComponentFixture<TutorialGroupFormComponent>;
+    let tutorialGroupFormComponent: TutorialGroupFormComponent;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ReactiveFormsModule, FormsModule, NgbTypeaheadModule],
+            declarations: [TutorialGroupFormComponent, MockPipe(ArtemisTranslatePipe)],
+            providers: [
+                MockProvider(CourseManagementService, {
+                    getAllUsersInCourseGroup: (courseId: number, courseGroup: CourseGroup) => {
+                        return of(new HttpResponse({ body: [] }));
+                    },
+                }),
+                MockProvider(AlertService),
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                tutorialGroupFormComponentFixture = TestBed.createComponent(TutorialGroupFormComponent);
+                tutorialGroupFormComponent = tutorialGroupFormComponentFixture.componentInstance;
+            });
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('should initialize', () => {
+        tutorialGroupFormComponentFixture.detectChanges();
+        expect(tutorialGroupFormComponent).not.toBeNull();
+    });
+
+    it('should not submit a form when title is missing', () => {
+        tutorialGroupFormComponentFixture.detectChanges();
+        tutorialGroupFormComponent.titleControl!.setValue(' ');
+
+        tutorialGroupFormComponentFixture.detectChanges();
+        expect(tutorialGroupFormComponent.form.invalid).toBeTrue();
+
+        const submitFormSpy = jest.spyOn(tutorialGroupFormComponent, 'submitForm');
+        const submitFormEventSpy = jest.spyOn(tutorialGroupFormComponent.formSubmitted, 'emit');
+
+        const submitButton = tutorialGroupFormComponentFixture.debugElement.nativeElement.querySelector('#submitButton');
+        submitButton.click();
+
+        return tutorialGroupFormComponentFixture.whenStable().then(() => {
+            expect(submitFormSpy).toHaveBeenCalledTimes(0);
+            expect(submitFormEventSpy).toHaveBeenCalledTimes(0);
```

Comment:
```suggestion
            expect(submitFormEventSpy).not.toHaveBeenCalled();
```

Reply:
Done in 26527db8ac4e1b72878a831b3e1e749b67b59361